rest-api/dao/questionnaire_response_dao.py

```diff
@@ -83,15 +83,51 @@ def _validate_model(self, session, obj):  # pylint: disable=unused-argument
       logging.error(
           'QuestionnaireResponse model has no answers. This is harmless but probably an error.')
 
+  def _validate_link_ids_from_resource_json_group(self, resource, link_ids):
+    """
+    Look for question sections and validate the linkid in each answer. If there is a response
+    answer link id that does not exist in the questionnaire, then raise an exception.
+    This is a recursive function because answer groups can be nested.
+    :param resource: A group section of the response json.
+    :param link_ids: List of link ids to validate against.
+    """
+    # look for child group sections and call ourselves
+    if 'group' in resource:
+      self._validate_link_ids_from_resource_json_group(resource['group'], link_ids)
+
+    if type(resource) is list:
+      for item in resource:
+        self._validate_link_ids_from_resource_json_group(item, link_ids)
+
+    # look for question section:
+    if 'question' in resource:
+
+      # if question is a list, then loop through the list
```

Comment:
nit: I like the documentation in this PR, the codebase could definitely use more. IMO some of these comments (like this one) could probably be removed though. In my experience inline comments tend to be useful when they target the "why" or the high level flow, but become less useful (or harmful) as they approach a restatement of the code (those generally lead to bloat and things going out of sync). I think this one is in the latter category.